descartes_samplers/include/descartes_samplers/samplers/impl/axial_symmetric_variable_offset_sampler.hpp

```diff
@@ -0,0 +1,148 @@
+ * limitations under the License.
+ */
+#ifndef DESCARTES_SAMPLERS_SAMPLERS_IMPL_AXIAL_SYMMETRIC_VARIABLE_OFFSET_SAMPLER_HPP
+#define DESCARTES_SAMPLERS_SAMPLERS_IMPL_AXIAL_SYMMETRIC_VARIABLE_OFFSET_SAMPLER_HPP
+
+#include "descartes_samplers/samplers/axial_symmetric_variable_offset_sampler.h"
+#include <iostream>
+
+const static std::size_t opw_dof = 6;
+
+namespace descartes_light
+{
+template <typename FloatType>
+AxialSymmetricVariableOffsetSampler<FloatType>::AxialSymmetricVariableOffsetSampler(
+    const Eigen::Transform<FloatType, 3, Eigen::Isometry>& tool_pose,
+    const typename KinematicsInterface<FloatType>::Ptr robot_kin,
+    const FloatType radial_sample_resolution,
+    const FloatType z_offset_max,
+    const FloatType z_offset_increment,
+    const typename CollisionInterface<FloatType>::Ptr collision,
+    const bool allow_collision)
+  : tool_pose_(tool_pose)
+  , kin_(robot_kin)
+  , collision_(collision)
+  , radial_sample_res_(radial_sample_resolution)
+  , allow_collision_(allow_collision)
+  , z_offset_max_(z_offset_max)
+  , z_offset_increment_(z_offset_increment)
+{
+}
+
+template <typename FloatType>
+bool AxialSymmetricVariableOffsetSampler<FloatType>::sample(std::vector<FloatType>& solution_set)
+{
+  std::vector<FloatType> buffer;
+
+  const auto nSamplesInBuffer = [](const std::vector<FloatType>& v) -> std::size_t { return v.size() / opw_dof; };
+
+  FloatType z_offset_current = static_cast<FloatType>(0.0);
+
+  while (solution_set.empty() && z_offset_current <= z_offset_max_)  // continue until solutions have been found or the
+                                                                     // maximum offset has been reached
+  {
+    FloatType angle = static_cast<FloatType>(-1.0 * M_PI);
+    while (angle <= static_cast<FloatType>(M_PI))  // loop over each waypoint
+    {
+      Eigen::Transform<FloatType, 3, Eigen::Isometry> p =
+          tool_pose_ *
+          Eigen::Translation<FloatType, 3>(Eigen::Matrix<FloatType, 3, 1>::UnitZ() *
+                                           static_cast<FloatType>(z_offset_current)) *
+          Eigen::AngleAxis<FloatType>(angle, Eigen::Matrix<FloatType, 3, 1>::UnitZ());
+
+      kin_->ik(p, buffer);
+
+      const auto n_sols = nSamplesInBuffer(buffer);
+      for (std::size_t i = 0; i < n_sols; ++i)
+      {
+        const auto* sol_data = buffer.data() + i * opw_dof;
+        if (isCollisionFree(sol_data))
+          solution_set.insert(end(solution_set), sol_data, sol_data + opw_dof);
+      }
+      buffer.clear();
+
+      angle += radial_sample_res_;
+    }  // redundancy resolution loop
+
+    z_offset_current += z_offset_increment_;
+  }
+
+  if (solution_set.empty() && allow_collision_)
+    getBestSolution(solution_set);
+
+  return !solution_set.empty();
+}
+
+template <typename FloatType>
+bool AxialSymmetricVariableOffsetSampler<FloatType>::isCollisionFree(const FloatType* vertex)
+{
+  if (collision_ == nullptr)
+    return true;
```

Comment:
Is the desired behavior to return that a trajectory is collision free if the user doesn't provide a means to check it? It seems like we should return false (or at least check `collision_allowed_`) because we weren't able to check it